ui/src/app/controls/endpoints/endpoints.component.ts

```diff
@@ -30,20 +30,23 @@ export class EndpointsComponent implements ControlValueAccessor, Validator {
   form = new FormArray<FormGroup>([]);
 
   constructor() {
+    this.form.markAllAsTouched();
```

Comment:
If I understand correctly, you are adding this to make all the fields with validation marked as red from the beginning. I don't think it is idiomatic to Material Design, where required fields are marked with (*), and validation of fields with specific pattern are considered when you are typing into them only

Reply:
That was initially a temporary way for me to quickly see at a glance if validation worked correctly and which fields were invalid or required.
But indeed, you're right. From the material design guidelines: `Show error text only after user interaction with a field`.
I'll revert the related commit soon.
Thanks.